activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/security/ProcessInstanceRestrictionService.java

```diff
@@ -27,24 +30,40 @@
 
     private ProcessDefinitionKeyBasedRestrictionBuilder restrictionBuilder;
 
+    private SecurityManager securityManager;
+
     public ProcessInstanceRestrictionService(SecurityPoliciesManager securityPoliciesManager,
                                              ProcessInstanceFilter processInstanceFilter,
-                                             ProcessDefinitionKeyBasedRestrictionBuilder restrictionBuilder) {
+                                             ProcessDefinitionKeyBasedRestrictionBuilder restrictionBuilder,
+                                             SecurityManager securityManager) {
         this.securityPoliciesManager = securityPoliciesManager;
         this.processInstanceFilter = processInstanceFilter;
         this.restrictionBuilder = restrictionBuilder;
+        this.securityManager = securityManager;
     }
 
     public Predicate restrictProcessInstanceQuery(Predicate predicate,
                                                   SecurityPolicyAccess securityPolicyAccess) {
+        Predicate initiatorPredicate = applyInitiatorRestriction(predicate);
+
         if (!securityPoliciesManager.arePoliciesDefined()) {
-            return predicate;
+            return initiatorPredicate;
         }
 
-        return restrictionBuilder.applyProcessDefinitionKeyFilter(predicate,
+        return restrictionBuilder.applyProcessDefinitionKeyFilter(initiatorPredicate,
                                                  securityPolicyAccess,
                                                  processInstanceFilter);
     }
 
+    private Predicate applyInitiatorRestriction(Predicate predicate) {
+        String userId = securityManager.getAuthenticatedUserId();
+
+        if (userId == null) {
```

Comment:
I'm not sure if there could be a scenario where the current user is null. I followed the same check here --> https://github.com/Activiti/activiti-cloud/blob/develop/activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/security/TaskLookupRestrictionService.java#L102